src/QMCWaveFunctions/Fermion/MatrixDelayedUpdateCUDA.h

```diff
@@ -431,18 +420,17 @@ class MatrixDelayedUpdateCUDA
     }
   }
 
-  Value* getRow_psiMinv_offload(int row_id) { return psiMinv_.device_data() + row_id * psiMinv_.cols(); }
-
   // prepare invRow and compute the old gradients.
   template<typename GT>
   static void mw_evalGrad(const RefVectorWithLeader<This_t>& engines,
+                          const RefVector<DualMatrix<Value>>& psiMinv_refs,
                           const std::vector<const Value*>& dpsiM_row_list,
                           const int rowchanged,
                           std::vector<GT>& grad_now)
   {
     auto& engine_leader = engines.getLeader();
     if (!engine_leader.isSM1())
-      mw_prepareInvRow(engines, rowchanged);
+      mw_prepareInvRow(engines, psiMinv_refs, rowchanged);
```

Comment:
this seems like a great chance to rename isSM1 since I immediately had to search for what it meant.

Reply:
I also would like to change it. It will happen when I rename MatrixDelayedUpdateCUDA class. Cannot do it on this PR due to complicating following PRs. Use https://github.com/QMCPACK/qmcpack/issues/4996 for tracking.

Reply:
Sounds like a good approach.